### PR TITLE
Update processor to typescript - Closes #5191

### LINKED ISSUE #5191: Update application/node/processor to be typescript
### Description

Update framework application/processor to be typescript

### Motivation

We gradually need to convert all the framework code into typescript

### Acceptance Criteria

- All files under framework/src/application/processor should be in typescript
- All related test should be in typescript

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #5191 

### How was it solved?

Update processor and block processor to typescript

### How was it tested?

- Check if the application runs properly with built version
- Test and build should pass

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -31,7 +31,11 @@ module.exports = {
 		'@typescript-eslint/no-dynamic-delete': 'off',
 		'@typescript-eslint/no-implied-eval': 'off',
 		'@typescript-eslint/strict-boolean-expressions': 'off',
+		'@typescript-eslint/no-unsafe-member-access': 'off',
+		'@typescript-eslint/no-unsafe-assignment': 'off',
+		'@typescript-eslint/no-unsafe-return': 'off',
```
Comment (nazarhussain) [inline .eslintrc.js:36]: I suggest we should not return `any` from a function we define.

Comment (shuse2) [reply .eslintrc.js:36]: yea I agree, but this check was checking more than that. 
Even with setup, it's still disallow `any`

Comment (nazarhussain) [reply .eslintrc.js:36]: Can't understand even with setup. Can you share any example which was causing problem more than function return.

Comment (shuse2) [reply .eslintrc.js]: seems like i was looking at the wrong error at that time, i cannot reproduce. Reverted the change =)

**Pull request merged**